capi-matrix/README.md

```diff
@@ -0,0 +1,78 @@
+# Cluster API Kubernetes/OS version management
+
+## Introduction
+
+We have a cluster-PROVIDER app for every provider that we support (it is our interface to define the cluster structure and configuration). It has a set of templates that allow our customers to create a cluster app with all necessary resources required by a Cluster API implementation. 
+
+In the values of these charts, among all possible parameters we define the Kubernetes version of the cluster and the machine image (AMI) that will be used to run the VMs. These images contain the Kubernetes components with a specific version. There are usually three AMIs defined in every provider: bastion, control-plane and node pools.
+
+This RFC born out of necessity to automate the cluster upgrades. For that reason, team Rocket has discussed and found a solution adding a Kubernetes and OS version in each chart release. Today AWS, GCP anf Open Stack providers pinned the Kubernetes/OS version in the chart. Here we propose an extension to give more flexibility and control to Giant Swarm and the customers.
+
+## User stories
+
+- As a customer, I don't want to worry about the kubernetes version or OS version used by my cluster machines. Just rely on the Giant Swarm default and tested version.
+
+- As a customer, I would like to choose the kubernetes version of my cluster without think which OS AMI correspond to that version.
```

Comment:
this story directly opposes the first story above

Reply:
yeah it is on purpose. Unfortunately, not every customer has the same mindset. Some of them want not to worry at all and some want to have finer control. I can add as `advance` customer WDYT?

Reply:
My issue with this is that we have not decided to give customers such freedom for now. The current product officially supports 2 Kubernetes minor versions at each point in time. Patch versions are not something we want to give freedom on, I would also very much question the use case here, a patch version in K8s should only fix problems and CVEs, there should be nothing speaking against it.

Furthermore, even within the 2 versions our product offering is currently:
- new features and minor changes only get into the latest
- the older version only gets patch updates

On the OS side we do not even give that choice and I would also like to keep that product decision unless there's a concrete use cases that makes sense here.

Exceptions to the above I know of, but I'd like to keep those to exceptions and not make them a feature.

TLDR I would reduce it to `As a customer, I would like to choose between the two supported minor Kubernetes versions without having to think about choosing an OS or machine image.`